schemas/pa-yaml/v3.0/pa.schema.yaml

```diff
@@ -401,24 +401,41 @@ definitions:
   DataSource-name:
     $ref: "#/definitions/entity-name"
 
-  DataSource-Type:
-    enum:
-      - DataverseTable
-
   DataSource-instance:
     type: object
-    required: [Type]
+    required:
+      - Type
+    additionalProperties: false
     properties:
-      Type: { $ref: "#/definitions/DataSource-Type" }
+      Type:
+        type: string
+        enum:
+          - Table
+          - Actions
+      ConnectorId:
+        type: string
+        nullable: true
+      Parameters:
+        type: object
+        additionalProperties:
+          type: string 
     oneOf:
       - required: [Type]
         additionalProperties: false
         properties:
           Type:
-            const: DataverseTable
-          TableLogicalName:
-            type: string
-            minLength: 1
+            const: Table
+          Parameters:
+            type: object
+            additionalProperties: false
+            properties:
+              TableLogicalName:
```

Comment:
is this a required property? if so, add: `required: [TableLogicalName]` before line 430.

Reply:
It's only required when the Type is Table

Reply:
correct, which is what this line is doing.

Reply:
looks like the latest iteration is all correct.